server/src/lsp4clj/core.clj

```diff
@@ -102,7 +102,11 @@
 
   (^void didSave [_ ^DidSaveTextDocumentParams params]
     (future
-      (handle-notification params feature-handler/did-save handler))
+      (try
+        (handle-notification params feature-handler/did-save handler)
+        (catch Throwable e
+          (logger/error e)
+          (throw e))))
```

Comment:
Usually, I don't like to log and throw, most of the time you want to do one or other, in this case, throwing would already make lsp4j log, right?

Reply:
Hum, not sure because the way we custom log, probably something we could fix at a higher level of lsp4clj, seems ok to keep this way for now

Reply:
Unfortunately, no. Usually you're right, when a thread has an uncaught exception it's automatically logged. But `future`s are special. They catch all exceptions and hold them, waiting to re-throw them when the future is dereffed (which we never do for this future.) So, since there aren't any uncaught exceptions, there's no automatic logging. I think (though I'm not sure) that if we changed `future` to `async/go` or `async/thread`, exceptions would be thrown.

You're right though, since this future is thrown away whether its body completes successfully or not, there's no real benefit to re-throwing. It'd be sufficient to log and return nil.

Reply:
Yeah, forgot this was a future, even so, as you said, logging in a future is enough indeed